### PR TITLE
Update Kubernetes upgrade documentation

### PR DESCRIPTION
**Pull request by briandowns**
Signed-off-by: Brian Downs <brian.downs@gmail.com>

#### Proposed Changes ####

Update Kubernetes upgrade documentation.

### CODE REVIEW

**Review by nikolaishields — APPROVED** [review]

File: developer-docs/upgrading_kubernetes.md [hunk 1]
```diff
@@ -58,8 +56,8 @@ Next, we need to create a release candidate (RC). The Drone (CI) process that bu
 * Click "Draft new release"
 * Select the target branch
 * Enter the desired version into the "Tag version" box. 
-    * Example tag: `v1.21.4+rke2r2`
-    * **NOTE** Make sure to create the tag against the correct release branch. In the example above, that would map to release branch `release-1.21`.
+    * Example tag: `v1.23.5-rc1+rke2r1`
+    * **NOTE** Make sure to create the tag against the correct release branch. In the example above, that would map to release branch `release-1.23`.
```
Comment (jossemargt-3pillar) [inline developer-docs/upgrading_kubernetes.md:60]:
```suggestion
    * Example tag: `v1.23.8-rc1+rke2r1`
    * **NOTE** Make sure to create the tag against the correct release branch. In the example above, that would map to release branch `release-1.23`.
```

Comment (jossemargt-3pillar) [inline developer-docs/upgrading_kubernetes.md:60]:
```suggestion
    * Example tag: `v1.22.8-rc1+rke2r1`
    * **NOTE** Make sure to create the tag against the correct release branch. In the example above, that would map to release branch `release-1.23`.
```

Comment (jossemargt-3pillar) [inline developer-docs/upgrading_kubernetes.md:60]:
```suggestion
    * Example tag: `v1.22.8-rc1+rke2r1`
    * **NOTE** Make sure to create the tag against the correct release branch. In the example above, that would map to release branch `release-1.22`.
```

Comment (jossemargt-3pillar) [reply developer-docs/upgrading_kubernetes.md:60]: Or change the release branch to `master`

Comment (jossemargt-3pillar) [reply developer-docs/upgrading_kubernetes.md:60]: As discussed offline `release-1.23` is fine for documentation purposes.

**Pull request merged**